src/osemgrep/language_server/scan_helpers/Scan_helpers.ml

```diff
@@ -64,56 +60,66 @@ let run_semgrep ?(targets = None) ?(rules = None) ?(git_ref = None)
 
 (** Scan all folders in the workspace *)
 let scan_workspace server =
-  let token =
-    create_progress server "Semgrep Scan in Progress" "Scanning Workspace"
-  in
-  let results, files = run_semgrep server in
-  Session.record_results server.session results files;
-  (* LSP expects empty diagnostics to clear problems *)
-  let diagnostics =
-    let files = Session.scanned_files server.session in
-    Diagnostics.diagnostics_of_results ~is_intellij:server.session.is_intellij
-      results files
+  let f () =
+    let token =
+      create_progress server "Semgrep Scan in Progress" "Scanning Workspace"
+    in
+    let results, files = run_semgrep server in
+    Session.record_results server.session results files;
+    (* LSP expects empty diagnostics to clear problems *)
+    let diagnostics =
+      let files = Session.scanned_files server.session in
+      Diagnostics.diagnostics_of_results ~is_intellij:server.session.is_intellij
+        results files
+    in
+    end_progress server token;
+    batch_notify server diagnostics
   in
-  end_progress server token;
-  batch_notify server diagnostics
+  wrap_with_detach f
```

Comment:
maybe a small comment here explaining why we need the detach, what are the Unix/Common blocking calls in f()
(I guess the call to 'run_semgrep' which internally use a lot Unix)